e2e/test/scaling/provider.go

```diff
@@ -74,26 +77,31 @@ func (p *Provider) findMachinePool(ctx context.Context) (*v1alpha3.MachinePool,
 
 	err := p.ctrlClient.List(ctx, crs, labelSelector, client.InNamespace(metav1.NamespaceDefault))
 	if err != nil {
-		return &v1alpha3.MachinePool{}, microerror.Mask(err)
-	}
-	if len(crs.Items) < 1 {
-		p.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("MachinePool CR for cluster id %q not found", p.clusterID))
-		return &v1alpha3.MachinePool{}, microerror.Maskf(notFoundError, fmt.Sprintf("MachinePool CR for cluster id %q not found", p.clusterID))
+		return []v1alpha3.MachinePool{}, microerror.Mask(err)
 	}
 
-	return &crs.Items[0], nil
+	return crs.Items, nil
 }
 
-func (p *Provider) AddWorker() error {
-	ctx := context.Background()
-	machinePool, err := p.findMachinePool(ctx)
+func (p *Provider) AddWorker(ctx context.Context) error {
+	machinePools, err := p.findMachinePools(ctx)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	machinePool.Spec.Replicas = to.Int32Ptr(*machinePool.Spec.Replicas + int32(1))
+	if len(machinePools) < 1 {
+		p.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("MachinePool CR for cluster id %q not found", p.clusterID))
+		return microerror.Maskf(notFoundError, fmt.Sprintf("MachinePool CR for cluster id %q not found", p.clusterID))
+	}
+
+	machinePool := machinePools[0]
```

Comment:
I'm not sure if k8s api guarantees ordering. Since this is in test, it might make sense to sort the slice before picking up a specific one by index. Alternative option would be to pick it up by name.

Reply:
This is not important at all.
For the needs of the scaling test we can scale up and scale down *any* node pool and it should always work